src/System.Private.CoreLib/shared/System/SpanHelpers.T.cs

```diff
@@ -482,62 +567,84 @@ public static int LastIndexOfAny<T>(ref T searchSpace, T value0, T value1, int l
             Debug.Assert(length >= 0);
 
             T lookUp;
-            while (length >= 8)
+            if (default(T) != null || ((object)value0 != null && (object)value1 != null))
```

Comment:
Would this check result in any perf regression for non-reference types (primitive types other than byte, like char, etc.)?

Reply:
nope, should is all constants so should just remove the `if` and the body of the `else`